packages/core/contracts/strategies/polygon/aave-leveraged/PolygonAaveLeveragedBorrowing.sol

```diff
@@ -0,0 +1,388 @@
+        _fullDeleverage();
+    }
+
+    function resume() public onlyGovernance {
+        _unpause();
+    }
+
+    function rebalance(uint256 borrowRate_, uint256 borrowDepth_)
+        external
+        onlyVaultOrGovernance
+    {
+        require(borrowRate_ <= borrowRateMax, "!rate");
+        require(borrowRate_ != 0, "borrowRate is used as a divisor");
+        require(borrowDepth_ <= BORROW_DEPTH_MAX, "!depth");
+
+        _fullDeleverage();
+        borrowRate = borrowRate_;
+        borrowDepth = borrowDepth_;
+        _enterRewardPool();
+    }
+
+    function increaseHealthFactor() external onlyVaultOrGovernance {
+        (uint256 supplyBal, ) = supplyAndDebt();
+
+        // Only withdraw the 10% of the max withdraw
+        uint256 toWithdraw = maxWithdrawFromSupply(supplyBal).mul(100).div(10);
+
+        ILendingPool(lendingPool).withdraw(
+            underlying,
+            toWithdraw,
+            address(this)
+        );
+        ILendingPool(lendingPool).repay(
+            underlying,
+            toWithdraw,
+            INTEREST_RATE_MODE,
+            address(this)
+        );
+    }
+
+    function _claimReward() internal {
+        address[] memory assets = new address[](2);
+        assets[0] = aToken;
+        assets[1] = debtToken;
+        IAaveIncentivesController(incentivesController).claimRewards(
+            assets,
+            type(uint256).max,
+            address(this)
+        );
+    }
+
+    function _enterRewardPool() internal {
+        uint256 amountToInvest = underlyingBalanceInStrategy();
+
+        if (amountToInvest > 0) {
+            _giveAllowance();
+            ILendingPool(lendingPool).deposit(
+                underlying,
+                amountToInvest,
+                address(this),
+                0
+            );
+            if (amountToInvest > minLeverage) {
+                for (uint256 i = 0; i < borrowDepth; i++) {
+                    amountToInvest = amountToInvest.mul(borrowRate).div(100);
+                    ILendingPool(lendingPool).borrow(
+                        underlying,
+                        amountToInvest,
+                        INTEREST_RATE_MODE,
+                        0,
+                        address(this)
+                    );
+                    ILendingPool(lendingPool).deposit(
+                        underlying,
+                        amountToInvest,
+                        address(this),
+                        0
+                    );
+
+                    if (amountToInvest < minLeverage) {
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+    function _liquidateReward() internal {
+        uint256 reward = IERC20(rewardToken).balanceOf(address(this));
+        _swap(reward, rewardToken, swapRoutes[rewardToken][underlying]);
+    }
+
+    // Divide the supply with HF less 0.5 to finish at least with HF~=1.05
+    function maxWithdrawFromSupply(uint256 _supply)
```

Comment:
To keep consistency within the codebase, internal/private methods should be prefixed by an underscore (eg. `_maxWithdrawFromSupply`.